hunts/src/hunt-view.js

```diff
@@ -1,161 +1,141 @@
+  rows.sort(puzzleComparator);
+  return rows;
+}
 
-    _puzzleComparator(a, b) {
-        // Solved puzzles should appear below unsolved ones
-        if (a.status == "SOLVED" && b.status != "SOLVED") {
-            return 1;
-        } else if (b.status == "SOLVED" && a.status != "SOLVED") {
-            return -1;
-        }
-        // Feeders before metas
-        if (a.feeders.length == 0 && b.feeders.length > 0) {
-            return -1;
-        } else if (a.feeders.length > 0 && b.feeders.length == 0) {
-            return 1;
-        }
-        // Newer puzzles before old ones
-        // TODO: once creation times are added to puzzles, use those instead
-        return b.id - a.id;
-    }
+export const HuntViewMain = (props) => {
+  const [huntData, setHuntData] = React.useState(props);
+  const [puzzleData, setPuzzleData] = React.useState([]);
 
-    _processPuzzleData(puzzleData) {
-        const rowMap = {};
-        puzzleData.forEach(row => {
-           rowMap[row.id] = row;
-        });
-        puzzleData.forEach(row => {
-            if (row.feeders.length > 0) {
-                row.subRows = [];
-                row.feeders.forEach(subRowId => {
-                    row.subRows.push(rowMap[subRowId]);
-                });
-            }
-        });
-        puzzleData.forEach(row => {
-            if (row.subRows) {
-                row.subRows.sort(this._puzzleComparator);
-            }
-        });
-        const rows = Object.values(rowMap).filter(row => row.metas.length == 0);
-        rows.sort(this._puzzleComparator);
-        return rows;
-    }
+  React.useEffect(() => {
+    const huntApiUrl = `/api/v1/hunt/${props.huntId}`;
+    const puzzlesApiUrl = `/api/v1/hunt/${props.huntId}/puzzles`;
+    fetch(huntApiUrl)
+      .then((response) => {
+        if (response.status > 400) {
+          //TODO: Some error handling should go here
+          console.error("Hunt API failure", response);
+        }
+        return response.json();
+      })
+      .then((huntData) => {
+        setHuntData(huntData);
+      });
 
-    render() {
-        if (this.state.huntData) {
-            return (
-            <div>
-                <h1>{this.state.huntData.name} - All Puzzles</h1>
-                <PuzzleTable columns={this.columns} data={this.state.puzzleData} />
-            </div>
-            );
-        } else {
-            return <h1>Loading...</h1>;
+    fetch(puzzlesApiUrl)
+      .then((response) => {
+        if (response.status > 400) {
+          // TODO: error handling
+          console.error("Puzzles API failure", response);
         }
-    }
-}
+        return response.json();
+      })
+      .then((puzzleData) => {
+        setPuzzleData(processPuzzleData(puzzleData));
+      });
+  }, [props.huntId]);
+
+  if (huntData) {
+    return (
+      <div>
+        <h1>{huntData.name} - All Puzzles</h1>
+        <PuzzleTable columns={TABLE_COLUMNS} data={puzzleData} />
```

Comment:
could potentially pass a "loading" prop to PuzzleTable while puzzleData is loading